src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.spring.dao.impl;
+
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Role getByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            Query<Role> findByRoleName = session.createQuery(
+                    "FROM Role r WHERE RoleName = :name", Role.class);
```

Comment:
you have different [name](https://github.com/mate-academy/jv-spring-security-part-2/pull/359/files#diff-a7802b09453eb95308ca4793c90b1474ce37d5deb1ff89eb69c4c26ee8cb45f3:~:text=private%20RoleName-,roleName,-%3B) in you entity, does it work?